tools/releaser/main.go

```diff
@@ -0,0 +1,289 @@
+		{cmd: "go", args: []string{"mod", "vendor"}},
+		{cmd: "find", args: []string{"vendor", "-name", "BUILD.bazel", "-delete"}},
+	}
+	for _, c := range initialCommands {
+		if out, err := exec.CommandContext(ctx, c.cmd, c.args...).CombinedOutput(); err != nil {
+			fmt.Println(string(out))
+			return err
+		}
+	}
+
+	workspace, err := os.ReadFile("WORKSPACE")
+	if err != nil {
+		return err
+	}
+
+	workspaceBuffer := bytes.NewBuffer(workspace)
+	workspaceWithoutDirectives := new(bytes.Buffer)
+
+	if verbose {
+		fmt.Println("Preparing temporary WORKSPACE without gazelle directives.")
+	}
+	directiveStart, err := getWorkspaceWithouthDirectives(workspaceBuffer, workspaceWithoutDirectives)
+	if err != nil {
+		return err
+	}
+
+	// write the directive-less workspace and update repos
+	if err := os.WriteFile("WORKSPACE", workspaceWithoutDirectives.Bytes(), 0666); err != nil {
+		return err
+	}
+
+	if verbose {
+		fmt.Println("Running update-repos outputting to temporary file.")
+	}
+	cmd := exec.CommandContext(ctx, "bazel", "run", "//:gazelle", "--", "update-repos", "-from_file=go.mod", fmt.Sprintf("-to_macro=%s%%gazelle_dependencies", _tmpBzl))
+	if out, err := cmd.CombinedOutput(); err != nil {
+		fmt.Println(string(out))
+		return err
+	}
+
+	depsMaybeBuff := new(bytes.Buffer)
+	workspaceDirectivesBuff := new(bytes.Buffer)
+	// parse the resulting tmp.bzl for deps.bzl and WORKSPACE updates
+	if verbose {
+		fmt.Println("Parsing temporary bzl file to prepare deps.bzl and WORKSPACE modifications.")
+	}
+	if err = getBuffsFromTmp(depsMaybeBuff, workspaceDirectivesBuff); err != nil {
+		return err
+	}
+
+	// update deps
+	depsFile, err := os.ReadFile("deps.bzl")
+	if err != nil {
+		return err
+	}
+	existingDepsScanner := bufio.NewScanner(bytes.NewBuffer(depsFile))
+	newDepsBuffer := new(bytes.Buffer)
+	var afterSkylib bool
+	for existingDepsScanner.Scan() {
+		if bytes.Contains(existingDepsScanner.Bytes(), []byte("skylib")) {
+			afterSkylib = true
+		}
+
+		if afterSkylib && bytes.HasSuffix(existingDepsScanner.Bytes(), []byte("_maybe(")) {
+			break
+		}
+
+		if _, err := newDepsBuffer.Write(existingDepsScanner.Bytes()); err != nil {
+			return err
+		}
+		if _, err := newDepsBuffer.Write([]byte("\n")); err != nil {
+			return err
+		}
+	}
+	// just append the rest
+	if verbose {
+		fmt.Println("Writing new deps.bzl")
+	}
+	if _, err := newDepsBuffer.Write(depsMaybeBuff.Bytes()); err != nil {
+		return err
+	}
+	if err := os.WriteFile("deps.bzl", newDepsBuffer.Bytes(), 0644); err != nil {
+		return err
+	}
+
+	// update WORKSPACE
+	if verbose {
+		fmt.Println("Writing new WORKSPACE")
+	}
+	directivelessWorkspace := workspaceWithoutDirectives.Bytes()
+	newWorkspace := make([]byte, 0, len(directivelessWorkspace))
+	newWorkspace = append(newWorkspace, directivelessWorkspace[:directiveStart]...)
+	newWorkspace = append(newWorkspace, workspaceDirectivesBuff.Bytes()...)
+	newWorkspace = append(newWorkspace, directivelessWorkspace[directiveStart:]...)
```

Comment:
You can write to the file one part at a time without concatenating. It would have smaller memory footprint